src/graph/gk_ops.cc

```diff
@@ -0,0 +1,106 @@
+/*!
+ *  Copyright (c) 2020 by Contributors
+ * \file graph/gk_ops.cc
+ * \brief Graph operation implemented in GKlib
+ */
+
+#if !defined(_WIN32)
+#include <GKlib.h>
+#endif  // !defined(_WIN32)
+
+#include <dgl/graph_op.h>
+
+namespace dgl {
+
+#if !defined(_WIN32)
+
+namespace {
+
+gk_csr_t *Convert2GKCsr(CSRPtr csr, bool is_row) {
+  // TODO(zhengda) The conversion will be zero-copy in the future.
+  const aten::CSRMatrix mat = csr->ToCSRMatrix();
+  const dgl_id_t *indptr = static_cast<dgl_id_t*>(mat.indptr->data);
+  const dgl_id_t *indices = static_cast<dgl_id_t*>(mat.indices->data);
+
+  gk_csr_t *gk_csr = gk_csr_Create();
+  gk_csr->nrows = mat.num_rows;
+  gk_csr->ncols = mat.num_cols;
+  size_t nnz = csr->NumEdges();
+  auto gk_indptr = gk_csr->rowptr;
+  auto gk_indices = gk_csr->rowind;
+  size_t num_ptrs;
+  if (is_row) {
+    num_ptrs = gk_csr->nrows + 1;
+    gk_indptr = gk_csr->rowptr = gk_zmalloc(gk_csr->nrows+1, "gk_csr_ExtractPartition: rowptr");
+    gk_indices = gk_csr->rowind = gk_imalloc(nnz, "gk_csr_ExtractPartition: rowind");
+  } else {
+    num_ptrs = gk_csr->ncols + 1;
+    gk_indptr = gk_csr->colptr = gk_zmalloc(gk_csr->ncols+1, "gk_csr_ExtractPartition: colptr");
+    gk_indices = gk_csr->colind = gk_imalloc(nnz, "gk_csr_ExtractPartition: colind");
+  }
+
+  for (size_t i = 0; i < num_ptrs; i++) {
```

Comment:
if sizeof(gk_indptr[0]) == sizeof(indptr[0])
I think we can use memcopy here at least.

Reply:
at this point, they are different. In the future (when GKlib supports both int32 and int64), we'll use memory copy.

Reply:
Shall we do some assertion before ahead? otherwise cast a int64 to int32 is dangerous.

Reply:
sure. we can double check the integer types.

Reply:
actually, when i wrote code, i didn't do any casting here. The code doesn't even assume the integer type. It should work for all different combinations. In the future, we'll check the type and cast the integers when necessary.